docs/content.en/docs/references/connectors/network_drive.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Network Drive"
+weight: 30
+---
+# Network Drive Connector
+
+## Register Network Drive Connector
+
+```shell
+curl -XPUT "http://localhost:9000/connector/network_drive?replace=true" -d '{
+  "name" : "Network drive Connector",
+  "description" : "Scan and extract metadata from network shared files.",
+  "category" : "cloud_storage",
+  "icon" : "/assets/icons/connector/network_drive/icon.png",
+  "tags" : [
+    "filesystem",
+    "storage",
+    "web"
+  ],
+  "url" : "http://coco.rs/connectors/network_drive",
+  "assets" : {
+  "icons" : {
+    "default" : "/assets/icons/connector/network_drive/icon.png"
+    }
+  }
+}'
+```
+
+> Use `Network Drive` as a unique identifier, as it is a builtin connector.
+
+## Update coco-server's config
+
+Below is an example configuration for enabling the Network Drive Connector in coco-server:
+
+```shell
+connector:
+  network_drive:
+    enabled: true
+    queue:
+      name: indexing_documents
+    interval: 60s
+```
+
+### Explanation of Config Parameters
+
+
+| **Field**    | **Type**  | **Description**                                                                        |
+| ------------ | --------- |----------------------------------------------------------------------------------------|
+| `enabled`    | `boolean` | Enables or disables the network drive connector. Set to`true` to activate it.          |
+| `interval`   | `string`  | Specifies the time interval (e.g.,`60s`) at which the connector will check for updates. |
+| `queue.name` | `string`  | Defines the name of the queue where indexing tasks will be added.                      |
+
+## Use the Network Drive Connector
+
+The Network Drive Connector allows you to index data from your SMB/CIFS shares into your system. Follow these steps to set it up:
+
+### Configure Network Drive Client
+
+To configure your Network Drive connection, you'll need to provide the server details and credentials. This setup allows your application to securely authenticate with and access files within a specified network share.
+
+First, you'll need your server information and credentials:
+
+`endpoint`: This is the IP address and port of your SMB server (e.g., `192.168.1.100:445`).
+
+`share`: This is the name of the shared folder you want to access.
+
+`username`: The username for authenticating to the share.
+
+`password`: The password for the specified user.
+
+`domain`: (Optional) The domain for NTLM authentication, often `WORKGROUP`.
+
+Next, you can specify which folders and files to index:
+
+`paths`: An array of strings representing the subdirectories to scan within the share. If you want to scan the entire share, you can use `["."]` or `[""]`.
+
+`extensions`: An array of strings defining specific file extensions to include (e.g., `["pdf", "docx"]`). If this list is empty or omitted, all file types in the specified paths will be considered.
```

Comment:
> If this list is empty or omitted, all file types in the specified paths will be considered.

The implementation does not behave in this way. Looks like an empty config would force the scan() function to skip all the files?

```go
// Extension name not matched
if len(extMap) > 0 && !extMap[fileExt] {
    return nil
}
```